src/core/thread/dua_manager.cpp

```diff
@@ -33,51 +33,115 @@
 
 #include "dua_manager.hpp"
 
-#if (OPENTHREAD_FTD || OPENTHREAD_MTD) && OPENTHREAD_CONFIG_DUA_ENABLE
+#if OPENTHREAD_CONFIG_DUA_ENABLE || OPENTHREAD_CONFIG_TMF_RPOXY_DUA_ENABLE
 
 #include "common/code_utils.hpp"
 #include "common/instance.hpp"
 #include "common/locator-getters.hpp"
 #include "common/logging.hpp"
 #include "common/settings.hpp"
 #include "net/ip6_address.hpp"
+#include "thread/mle_types.hpp"
 #include "thread/thread_netif.hpp"
+#include "thread/thread_tlvs.hpp"
+#include "thread/thread_uri_paths.hpp"
 #include "utils/slaac_address.hpp"
 
 namespace ot {
 
 DuaManager::DuaManager(Instance &aInstance)
     : InstanceLocator(aInstance)
+    , Notifier::Receiver(aInstance, DuaManager::HandleNotifierEvents)
+    , mTimer(aInstance, DuaManager::HandleTimer, this)
+    , mRegistrationTask(aInstance, DuaManager::HandleRegistrationTask, this)
+    , mDuaNotification(OT_URI_PATH_DUA_REGISTRATION_NOTIFY, &DuaManager::HandleDuaNotification, this)
+    , mIsDuaPending(false)
+#if OPENTHREAD_CONFIG_DUA_ENABLE
+    , mDuaState(kNotExist)
+    , mLastRegistrationTime(0)
     , mDadCounter(0)
+#endif
+#if OPENTHREAD_CONFIG_TMF_RPOXY_DUA_ENABLE
+    , mChildIndexDuaRegistering(0)
+    , mRegisterCurrentChildIndex(false)
+#endif
 {
+    mDelay.mValue = 0;
+
+#if OPENTHREAD_CONFIG_DUA_ENABLE
     mDomainUnicastAddress.Clear();
     mDomainUnicastAddress.mAddressOrigin      = OT_ADDRESS_ORIGIN_THREAD;
-    mDomainUnicastAddress.mPreferred          = true;
+    mDomainUnicastAddress.mPreferred          = false;
     mDomainUnicastAddress.mValid              = true;
     mDomainUnicastAddress.mScopeOverride      = Ip6::Address::kGlobalScope;
     mDomainUnicastAddress.mScopeOverrideValid = true;
 
     mFixedDuaInterfaceIdentifier.Clear();
+#endif
+
+#if OPENTHREAD_CONFIG_TMF_RPOXY_DUA_ENABLE
+    memset(&mChildDuaMask, 0, sizeof(mChildDuaMask));
```

Comment:
Not necessary for this PR, how about adding a method to `ChildMask` to clear it?

Reply:
thanks. just added in this PR as it is a minor change.